### ISSUE TITLE
[Suggestion] Half Block - Why it must stay

### ISSUE THREAD

**Issue by SabreMania** [tsunamayo/Starship-EVO#1648]
Greetings. I don't post on the Github often, because I don't quite play enough to find a ton of bug (With that said, I have found a few, though they have all been already mentionned by others such as Dwarf-LordPangolin). But I am not here to talk about bugs, I am here to voice my concerns over the half-block's seemingly unavoidable removal.

I come with many voices behind me. A great deal of the community agrees with my following position, that the Half-Block is an incredible asset to this game, and that bricks cannot quite replace it. 

From what I was told, the decision that motivates you to remove it is a combinaison of problems related to AI Pathfinding, and supposedly, that you believe that this block is not popular or widely used, which, if true, I could not disagree more.

Every build that I have begun, I have used it for various visual or fundamental aspects of the design, in ways that bricks could not replicate, or could, but at the cost of game performance (a large number of bricks is bound to cause more lag than a smaller, equivalent number of blocks), as well as the cost of time. My best example being this, my latest build:
![snekpek2](https://user-images.githubusercontent.com/52141971/76168583-8b8bf900-6147-11ea-81e6-86ba61cad572.png)

![snekpek](https://user-images.githubusercontent.com/52141971/76168540-35b75100-6147-11ea-9c90-78eaa3d8d6ca.png)

This particular 'path' of blocks and halfblocks stretches across 81 blocks for the length alone, and is 5 blocks tall. It took me around 10 minutes to build it, and thanks to the lovable mirror feature, the other side is perfectly identical. 

What this build showcases is something that I plan to do very often, as well as the utility of halfblocks in general. They can be used to make both chaotic, obvious elements, as well as smooth and subtle details quickly and efficiently. While a better final result -could- potentially be achieved by using bricks, for someone like me who puts function over aesthetics, this block is one of many perfect compromises. 

I could replicate the entirety of this pattern with bricks. But the time. The time it would take would be well above four times as long. I would spend an hour or more, and I would still not be done, due to the lengthy process of building that this game operates on. It makes me miss Starmade's efficient build-mode, but I digress.

As I mentionned above, I am not the only one who uses the half-block. In fact, I don't think, amongst the active community, that there is a single person who does not uses it in their builds, save for builds that are aimed towards being implemented in your game, since you specified it not be included due to you wishing to remove it.

But I still cannot stress enough the importance of such a block. When I say that I use it in 'every' build I have, I mean it. Even my large turrets have them, because they are way too conveniant at putting details that are -not- made out of bricks. Many blocks were removed that I found very useful, such as the block version of the window tile edge, which I too, miss. But unlike these blocks, the half-block is still ingame, and I would wish for it to remain as well.

Very few people are active on the github, and as such it might be difficult to gauge just how many people are actually using the block. But please, take me desperately coming on here to tell you not to remove it, as me representing the countless others who do not come here, but secretly beg you likewise not to take it away. 

Bricks are amazing, I agree. They make for a unique system that allows this game to have more details on it's ships than any other game out there. But I don't believe this is a reason to cut into useful block shapes, and I certainly don't believe you want people to run around with massive capital ships that have either very little detail, or an absurd amount of bricks used to put this detail onto them.

With all due respect, I hope this post will make you reconsider your decision to remove the half-block. 

-A fan who loves your game.

**Comment by PopkaAkula** [comment]
![unknown](https://user-images.githubusercontent.com/37738021/76169023-d4a67200-6174-11ea-9bde-4cbca8f4747e.png)

One reason as to why the slab should be kept is because of this, on the left you have a slab on top of a block, and on the right you have a brick as a slab on top of a block. The slab blends with the block and matches the "style" of the block whereas the brick **doesn't** blend and **doesn't** match the style of the block. If you remove block slabs, this issue will be even more prevalent and will prevent people from making their ships look the way they want.

**Comment by ProPeach** [comment]
I'd just like to bring up all the points made by testers across #1612 and #1626 . We feel very strongly about the slab issue and other similar issues like the removal of block canopies

I don't build many large scale things so I have few examples of my own. I've highlighted the slabs in red, here they're used to accurately increase thickness of a cuboid, and to enable me to build an even number thickness part on a build with odd number symmetry which is incredibly useful. Yes, I could replace them with bricks, but as other users have pointed out they won't blend thanks to the brick border, it will take much longer to place as bricks and they'll use more computing resources as I'll have to place more of them. That part of the ship will also have less HP if I used bricks rather than blocks too which would discourage a player to use them in combat oriented ships.

![image](https://user-images.githubusercontent.com/27857711/76169550-349b1980-6171-11ea-990a-3b67fd141d47.png)

**Comment by tsunamayo** [comment]
First thanks for your input! Thanks also for speaking from other people in the community.
First I know people use this block, I have been using it myself too much! 
I dont want to get rid of it not because I think people are not using it, I want to get rid of it because I think people are using it a lot for floor and ceiling. I agree it is not the only use, but that seems like one of the most obvious one (and what I have been doing), maybe because of minecraft, and also because 2.5 m seems the right size for a room, you save space ect... Tell me if you agree with that.

From your contribution I understand there is also: 
 - Fast Gribbling and Detailing
 - Accurately increase thickness of a cuboid (ty @ProPeach ;) )
 
Now the problem is that it make my life really tricky for lighting, and in the future air pressure and AI pathfinding (being used for interior block). It get really extra tricky when you put a light into the other half space of the block, which is very likely to happen if you use it for ceiling / floor (I have been doing that myself a lot!) I will have to do a light refactor in the future as the current model dont deal well with reflection, shadow and have a big lag when building on large ship, but that would be post early access.

I would like to push player into making standardized 1m increment height for interior.
Now from my experience people get sad when I remove stuff and they are 100% right, player dont have to see the technical side being that (thats my job!).
So basically as I want to hit early access soon and open it for ship rewards, I want to do my cleanup now so I dont have to do it in the future, to not piss off even more people!

I know that sucks, but if my tech can handle it in the future (which it might), I will bring them back.

Also what starmade features would you like to see (I didnt had time to play starmade). I want to target something a bit more simple but I am always open for suggestion.

**Comment by SabreMania** [comment]
As I slightly suspected, you wish to remove the block due to technical engine problems, which I believe is perfectly understandable.

With that being said, this is such a useful block that I must fight for it to remain, and to do so, I will provide with an alternative solution. You say you wish to remove it so that people won't use it to make floors and ceilings. But perhaps in this situation, a good compromise would be the following:

Modifying the hitbox of the block. I am not quite sure how you have handled your game's hitboxes, but If you were to make it's hitbox the same as a 1 meter block, it would heavily discourage it's usages in the form that you wish people not to use it as, while not affecting it's usages as a gribble-block. With that being said, removing it would not mean that players would be encouraged to build in 1 meter block increments, as they could simply use bricks in their builds, except for builds for the official game where you would specify a specific height for your rooms and hallways. 

If that is not possible, or that it is not an acceptable solution, I don't see other solutions, other than removing it of course. But if I do see one i'll be sure to mention it in either another post, or modify this one.



Putting that aside, since you seem open to suggestions regarding Starmade features, I would like to suggest something akin to it's own build mode!

In Starmade, the way you build was slightly different than what we currently have in the game. Rather than fly around your ship and place blocks one at a time (or stretch them to a maximum length, width and height of 16 meters), you entered your ship's 'core' and then had a free-floating camera with instant acceleration (so the moment you stopped pressing W, your camera would immediately stop, rather than have that little bit of lag-behind that we have here). 

It also had a 'slow-move' option where your camera would move at a decently slow speed, and a 'fast move' where your camera would drastically accelerate, which was perfect for when you had big builds that required you to move from one end of your ship to the other. The two modes were togglables with the press of a button that you had to hold down to stay in the 'fast move' mode. 

But the biggest advantages I believe, were the following : The fact that your camera could go straight through blocks, and go inside of your ship, which is not possible here. This allowed a player to work on the outside, or on the inside, and switch between any of the two at any given time with very little time wasted on switching. 

And the fact that you could, in some kind of 'advanced build mode', modify entirely how many blocks you could place down in a single click of a button (if you stayed in the advanced build mode). This allowed you to place 300 blocks every time you clicked with your mouse, in any axis that you want, if not two or three at once, if you so desired, and then you could switch back to 'simple' build mode and go back to placing single blocks, one at a time, without losing time on re-setting your 'advanced build mode' 's massive box-clicker.

Of course this wouldn't quite work with the way this game is designed. The last part at least. However, I believe something very similar could be done, where in some kind of 'advanced building' mode, the game would save any shape of 'stretched' blocks. So for example, if you want to quickly place down increments of 1 normal block that is stretched in a 16x5x3 segment, you could set it as such in your advanced building tool, save it, and then just click away and instantly place down those 'pre-stretched' blocks. It would drastically increase building speed, especially if you could switch back and forth between this advanced building mode and the current building mode we have in game.

But of course, I have no idea how hard this would be to code, or how long it would take, so I am not saying it should be done right away. But on the other hand it -would- mitigate the loss of the half-block as well as any other block that would be removed from the game (and that could be replaced by using bricks).

**Comment by Garrett-C** [comment]
One large issue I see is that removing the slab will just cause people to do the same thing with bricks this is something I know you are equally opposed to. When you look at the difference in size limits I think it is fair to say by removing the half block you will ultimately hurt performance. Half blocks can be used to fill a 16x16 space, bricks can only do 4x4. So by removing the half block you will make it that people need to use multiple bricks.

**Comment by ProPeach** [comment]
Thanks for taking the time to reply in full, it's very valuable to know the full picture from both sides.
I completely agree that a smooth, frustration free experience is essential for the game's release. New players would only have the positive experience of you adding blocks without knowing that happened before Early Access with the removal of some shapes and block types.

However, I don't think removing the slab will solve the problems you mention. Bricks are an extremely attractive alternative for these shapes, and players will use definitely use them as you say. But bricks behave even worse with lighting than slabs, with light leaking through them and still enabling the placement of lights in half meter spaces as you mention. Players are also more likely to use bricks in their corridor walls, floor and ceiling without the slabs; as I understand it solving the lighting and pathfinding problems for bricks would be far more difficult than to solve them for block slabs where they conform to a nice 1m grid. Personally I think it would be a better investment to fix the issues slabs face, at least with lighting now rather than release the game in a state that would encourage the use of bricks which are worse for lighting, pathfinding and performance.

**Comment by tsunamayo** [comment]
If people wants to use brick for their interior at this point there is nothing I can do, I already made them inefficient vs hull block. 
If I manage to indeed fix the issue with slabs then I will put them back! I cant do it now as pathfinding and the new light system will be post early-access.
Your feedback is exactly the reason why I need to remove them now, and not after early access... To piss-off less people...
Thanks guy and sorry to have changed so many things...
